src/layer/poi.js

```diff
@@ -0,0 +1,81 @@
+import * as label from "../constants/label.js";
+import * as Color from "../constants/color.js";
+
+const poiIcon = [
+  "match",
+  ["get", "class"],
+  ["bar", "beer"],
+  "poi_martini_glass",
+  "hospital",
+  "poi_health_cross",
+  "parking",
+  "poi_p",
+  "poi_square_dot", //icon for generic POI, not currently used
+];
+
+const poiFilter = [
+  ">=",
+  ["zoom"],
+  [
+    "match",
+    ["get", "class"],
+    "hospital",
+    15,
+    ["bar", "beer"],
+    16,
+    "parking",
+    17,
+    18,
+  ],
+];
+
+const iconColor = [
+  "match",
+  ["get", "class"],
+  ["bar", "beer"],
+  Color.poi.consumer,
+  ["hospital", "parking"],
+  Color.poi.infrastructure,
+  Color.poi.infrastructure,
+];
+
+const labelPaint = {
+  "text-halo-color": "#ffffff",
+  "icon-halo-color": "#ffffff",
+  "text-halo-width": 1.5,
+  "icon-halo-width": 0.4,
+  "text-halo-blur": 1,
+  "icon-halo-blur": 0.2,
+  "text-color": iconColor,
+};
+
+export const poi = {
+  id: "poi",
+  type: "symbol",
+  paint: labelPaint,
+  filter: poiFilter,
+  layout: {
+    "text-font": ["Metropolis Light"],
+    "text-optional": true, //TODO: disable this for generic (square dot) POIs
```

Comment:
Some POIs are sufficiently descriptive by their icons even when there isn’t enough room for the label. This includes community anchor institutions but also more mundane things like parking lots. But I’m not certain that it will always apply to every POI class with an icon. What about explicitly saying which classes can appear without a label?

```suggestion
    "icon-optional": ["match", ["get", "class"], ["hospital", "parking"], true, false],
```

/ref https://github.com/ZeLonewolf/openstreetmap-americana/pull/387#discussion_r895006271